module/vision/YoloCoco/src/YoloCoco.cpp

```diff
@@ -0,0 +1,185 @@
+
+            auto bounding_boxes       = std::make_unique<BoundingBoxes>();
+            bounding_boxes->id        = img.id;
+            bounding_boxes->timestamp = img.timestamp;
+            bounding_boxes->Hcw       = img.Hcw;
+
+            // Helper function to simplify unprojection calls
+            auto pix_to_ray = [&](double x, double y) {
+                // Normalize the pixel coordinates to the image width normalized dimensions
+                Eigen::Vector2d norm_dim = Eigen::Vector2d(img.dimensions.x(), img.dimensions.y()) / img.dimensions.x();
+                return unproject(Eigen::Matrix<double, 2, 1>(x / img.dimensions.x(), y / img.dimensions.x()),
+                                 img.lens,
+                                 norm_dim);
+            };
+
+            // Helper function to simplify projecting rays onto the field plane then transforming into camera space
+            auto ray_to_camera_space = [&](const Eigen::Matrix<double, 3, 1>& ray) {
+                Eigen::Vector3d uBCw = Hwc.rotation() * ray;
+                Eigen::Vector3d rPWw = uBCw * std::abs(Hwc.translation().z() / uBCw.z()) + Hwc.translation();
+                return Hwc.inverse() * rPWw;
+            };
+
+            for (size_t i = 0; i < indices.size(); i++) {
+                // Get the index of the detected object from list of indices
+                int idx = indices[i];
+                // Get the class id associated with the detected object
+                int class_id = class_ids[idx];
+
+                // Convert the bounding box points to unit vectors (rays) in the camera {c} space
+                Eigen::Vector3d top_left_ray  = pix_to_ray(boxes[idx].x, boxes[idx].y);
+                Eigen::Vector3d top_right_ray = pix_to_ray(boxes[idx].x + boxes[idx].width, boxes[idx].y);
+                Eigen::Vector3d bottom_right_ray =
+                    pix_to_ray(boxes[idx].x + boxes[idx].width, boxes[idx].y + boxes[idx].height);
+                Eigen::Vector3d bottom_left_ray = pix_to_ray(boxes[idx].x, boxes[idx].y + boxes[idx].height);
+                Eigen::Vector3d centre_ray =
+                    pix_to_ray(boxes[idx].x + boxes[idx].width / 2.0, boxes[idx].y + boxes[idx].height / 2.0);
+                Eigen::Vector3d bottom_centre_ray =
+                    pix_to_ray(boxes[idx].x + boxes[idx].width / 2.0, boxes[idx].y + boxes[idx].height);
+                Eigen::Vector3d top_centre_ray = pix_to_ray(boxes[idx].x + boxes[idx].width / 2.0, boxes[idx].y);
+
+                auto bbox        = std::make_unique<BoundingBox>();
+                bbox->name       = objects[class_id].name;
+                bbox->confidence = class_confidences[idx];
+                bbox->corners.push_back(top_left_ray);
+                bbox->corners.push_back(top_right_ray);
+                bbox->corners.push_back(bottom_right_ray);
+                bbox->corners.push_back(bottom_left_ray);
+
+                emit(std::move(bounding_boxes));
+
```

Comment:
```suggestion
                bbox->corners.push_back(bottom_left_ray);
                 bounding_boxes.push_back(bbox);
                 }
                emit(std::move(bounding_boxes));

```
I think this is what you want? but obv not formatted